src/deriv_api/DerivAPIBasic.js

```diff
@@ -253,16 +268,37 @@ export default class DerivAPIBasic extends DerivAPICalls {
         this.events.next({
             name: 'close',
         });
-
         if (this.shouldReconnect) {
             this.events.next({
                 name: 'reconnecting',
             });
 
+            this.reconnect();
+        }
+    }
+
+    /**
+     * Clears previous connection keeplive ping timeout and connect & assign the handles
+     */
+    reconnect() {
+        this.pong(); // clear all previous timeout
+        if (this.shouldReconnect) {
             this.connect();
+            this.connection.onopen    = this.openHandler.bind(this);
+            this.connection.onclose   = this.closeHandler.bind(this);
+            this.connection.onmessage = this.messageHandler.bind(this);
+            this.connection.onerror   = this.errorHandler.bind(this);
```

Comment:
connect() get called only when connect arguments are passed. if the connection made in outside this api & passed to construction then connect() is not called. If we move the handler assign part to connect()  then the handlers are not get called and responses are not returned if we make connection outside this api . I tried this out using small sample script. (for example: only connection created in third party side, they use default openhandler, messagehandler... in the api)

To avoid duplication, I created function connectionHandlers() where these handlers assigned to connection & called this function in both the places